ADR/0033-enable-native-opentelemetry-tracing.md

```diff
@@ -0,0 +1,39 @@
+# 33. Enable Native OpenTelemetry Tracing
+
+Date started: 2024-02-27
+
+Date accepted: 2024-04-02
+
+## Status
+
+Accepted
+
+## Context
+
+Konflux is a tool under active development and, therefore, unforeseen issues may arise. A recent (at the time of writing this ADR) example is the [long running](https://github.com/redhat-appstudio/build-definitions/pull/856/checks?check_run_id=22307468968) [e2e-test](https://github.com/redhat-appstudio/build-definitions/blob/main/.tekton/tasks/e2e-test.yaml) in Konflux’s build definitions. Fixing and debugging such issues is not a trivial thing for Konflux’s developers. Additional data, metrics, telemetry and tracing are essential in enabling Konflux developers and SREs to come up with fixes.
+
+Tracing, in particular, enables a straightforward model for dealing with complex, distributed systems. It gives unique insight into a system’s execution, grouping functions together. These grouping functions can be critical for finding fields that correlate to some problem, and provide powerful insights to reduce the range of possible causes.
+
+OpenTelemetry is the industry standard to instrument, generate, collect, and export telemetry data (metrics, logs, and traces) to help you analyze software performance and behavior. Our goal is to collect [traces](https://opentelemetry.io/docs/concepts/signals/traces/) from the Konflux activity of building applications.
+
+Instrumenting Konflux with [OpenTelemetry (OTel)](https://opentelemetry.io/docs/) tracing will provide SREs and developers invaluable insight for incident response, debugging and monitoring. Our goal is to get traces for the Konflux activity in order to achieve an easier mental model to use for debugging.
+
+## Decision
+
+We are going to enable as much native tracing in Konflux as we can by quickly enabling any pre-existing tracing capabilities in the system. Any other type of tracing (e.g. zero code instrumentation or code based instrumentation) is out of scope for this ADR.
+
+For instance, [Pipeline Service](https://github.com/redhat-appstudio/architecture/blob/main/architecture/pipeline-service.md) has to be instrumented with OTel as it provides Tekton APIs and services to Konflux. Pipeline Services includes Tekton which already natively supports [OpenTelemetry Distributed Tracing for Tasks and Pipelines](https://github.com/tektoncd/community/blob/main/teps/0124-distributed-tracing-for-tasks-and-pipelines.md), so no upstream changes are required. We just need to work to enable this native tracing (e.g. set environment variables).
```

Comment:
A bit misleading, in that the feature you note only applies to the core tekton controller.

Neither PAC, nor chains, nor results has the capability yet.  At least some in Konflux consider all those "Pipeline Service".

If you want this across all of tekton, you need to duplicate https://github.com/tektoncd/pipeline/pull/5746 in all those other projects.

Reply:
Of course with open source, everyone can contribute :-) .... but if we stay in silos and expect openshift pipelines to update those other projects, updating those other projects will get prioritized against a bunch of other Konflux and non-Konflux asks, as there is not longer a dedicated team for the Konflux pipeline service mission (with it getting rolled into openshift pipelines).

Reply:
Then I guess more accurate wording would be to state that we are enabling tracing in the core tekton controller in Konflux instead of the pipeline service then?

I'm asking because the proposed scope for this ADR is to enable as much native tracing as possible, and not add code-based instrumentation (that would be part of a follow up ADR).

Reply:
> Then I guess more accurate wording would be to state that we are enabling tracing in the core tekton controller in Konflux instead of the pipeline service then?

yes;  and then say upstream work is needed to enable this for pac, chains, and results

also, if possible, I'd like to start reducing our use of  the term "pipeline service" , and just say Tekton or OpenShift Pipelines (the RH Tekton offering).  At this point, all "pipeline service" really means is how Konflux configures Tekton.
> 
> I'm asking because the proposed scope for this ADR is to enable as much native tracing as possible, and not add code-based instrumentation (that would be part of a follow up ADR).

sounds good / makes sense .... that follow up notion, it not already detailed in the ADR, should be then.

Reply:
Updated the ADR. PTAL.

Reply:
realized my abbreviations may not have been expansive enough ... just in case:
- "pac" : pipelines as code i.e. https://github.com/openshift-pipelines/pipelines-as-code/ ... this is how Konflux receives github / gitlab events and triggers the secure supply chain pipelines
- "chains" : tekton chains i.e. https://github.com/tektoncd/chains which is leveraged by the enterprise contract team for all  the attestations, signings, etc.
- "results" : tekton results i.e. https://github.com/tektoncd/results which is how Konflux currently is offloading pipelinerun/taskrun yaml and underlying pods logs to RDS and S3 respectively so that users can access that data after those items are pruned from etcd, where we prune from etcd to both 1) avoid the scaling limits of etcd, 2) free up the PVCs currently used by the Konflux pipelines, as AWS limits on these are well beyond Konflux's needs